Add New Feature and Refactor Proxy Server

evan-m-jackson opened the pull request:
- Add Port configuration
- Refactor YAML to deploy only on pushes to main
- Refactor YAML to run tests and prevent merge if any tests fail

evan-m-jackson committed:
Merge pull request #15 from evan-m-jackson/remove-idea-file

Remove .idea file

evan-m-jackson committed:
Light Refactor of Code

- Delete 'HTTPServer.service' file
- Refactor 'Tar up the build step' in azure-vm.yml file
- Remove excess whitespace in IntegrationTest.cs file

evan-m-jackson committed:
Light Refactor of Code

- Delete 'HTTPServer.service' file
- Refactor 'Tar up the build step' in azure-vm.yml file
- Remove excess whitespace in IntegrationTest.cs file
- Format ResponseByPathTests.cs and ResponseTests.cs file

evan-m-jackson committed:
Refactor File Names and Paths

- Create new directories and renamed existing directories
- Organize project files in sub-directories
- Refactor tests to correspond with specific classes
- Organize test files in sub-directories
- Perform heavy formatting
- Move test doubles into test directories

evan-m-jackson committed:
Build Proxy Server Capabilities

- Add 'GetCode' method to 'Header' class to return status code
- Add if/else statement in Program class to allow for proxy server capabilities
- Add 'ProxyResponse' class and unit tests

evan-m-jackson committed:
Add PUT and DELETE capabilities to 'ProxyResponse' class

evan-m-jackson committed:
Refactor ProxyResponse class and tests

- Refactor GetBody method
- Remove unnecessary test

evan-m-jackson committed:
Rename namespaces to match file paths

evan-m-jackson committed:
Add ToDo First ID capabilities and format

- Refactor initialized variables to begin with '_'
- Remove unneccessary usings in each class
- Rename Proxy test files
- Add First ID class to return first todo item's primary id
- Remove empty whitespaces

evan-m-jackson committed:
Remove dictionay paramter default in ResponsePath class

evan-m-jackson committed:
Refactor FirstID class

- Create RunToListRequest method that runs the WriteToDoListRequest method until the body string reads the correct string, which is the JSON object with the to-do ids

evan-m-jackson committed:
Refactor Proxy Response Class

- Create 'GetContentLength' method that returns the length of the response body from the Proxy Server request and implemented this method into the 'GetResponse' method.
- Create 'StatusWrapper' method that returns the status code based on the HTTP method. This allowed for the 'GetResponse' method's design to be cleaned up.

evan-m-jackson committed:
Remove unnecessary print statements

evan-m-jackson committed:
Update README

- Add instructions for acceptance tests

evan-m-jackson committed:
Add New Feature and Refactor Proxy Server

- Create UpdateInitialLine class to handle updating path for the proxy server request's
- Create interface called IGetFirstID to double GetFirstID (formerly FirstID) into TestGetFirstID class for testing UpdateInitialLine class
- Create Port class to allow the user to choose the port they want to run the server on. Made it default to port 5000.
- Light formatting and removing whitespaces

evan-m-jackson committed:
Add New Feature and Refactor Proxy Server

- Create UpdateInitialLine class to handle updating path for the proxy server request's
- Create interface called IGetFirstID to double GetFirstID (formerly FirstID) into TestGetFirstID class for testing UpdateInitialLine class
- Create Port class to allow the user to choose the port they want to run the server on. Made it default to port 5000.
- Light formatting and removing whitespaces

evan-m-jackson committed:
Refactor Port class

- Create IsPortValid method to return a boolean if the port number is between the range of 0 to 65535.
- Create GetPortNum method to returns the port string as an int or 5000 as default

evan-m-jackson committed:
Refactor Port class

- Create IsPortValid method to return a boolean if the port number is between the range of 0 to 65535.
- Create GetPortNum method to returns the port string as an int or 5000 as default

evan-m-jackson committed:
Refactor Github workflow files

evan-m-jackson committed:
Refactor Github workflow files

evan-m-jackson committed:
Merge pull request #20 from evan-m-jackson/update-github-actions

Refactor Github workflow files

paytonrules commented on HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs:
```diff
@@ -0,0 +1,178 @@
+
+        var proxyResponse = new ProxyResponse(reader, writer, httpPath, httpType);
+        proxyResponse.GetResponse();
+		var actualStatusLine = stream[0];
```
Got an indentation issue here.

evan-m-jackson replied:
Agreed. Will update

paytonrules commented on HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs:
```diff
@@ -0,0 +1,178 @@
+    public void GetTodoPOSTValidResponseTest()
+    {
+        var expectedStatusLine = "HTTP/1.1 201 Created";
+        var proxyStream = new List<string>() { "HTTP/1.1 200 OK", "Content-Type: application/json", "Content-Length: 20", "", "{ 'task': 'First Item' }" };
```
These lists should broken up into multiple lines - they are really long.

evan-m-jackson replied:
Agreed. Will update

paytonrules commented on HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs:
```diff
@@ -0,0 +1,178 @@
+
+namespace HTTPServerProxyTests.Response;
+
+public class UnitTestsForProxyResponse
```
Nitpick here but the name should probably ProxyResponseTest or TestProxyResposne - so I can tell what class is being tested at a glance. And "UnitTestsFor" is a little long and unconventional.

evan-m-jackson replied:
Agreed. Will update

paytonrules commented on HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs:
```diff
@@ -0,0 +1,178 @@
+        var httpPath = "todo";
+        var writer = new TestWriteStreams(stream);
+
+        var proxyResponse = new ProxyResponse(reader, writer, httpPath, httpType);
```
Giving the number of constructor parameters here you might benefit from a Factory function or the Builder pattern. A less complex solution would be to use named parameters here. CleanCode uses a guideline of 2 parameters or less to any function, unless you can name them.

evan-m-jackson replied:
Sounds good. I think I'll just use named parameters

paytonrules commented on HTTPServerProject.Tests/HTTPServerProxyTests/ResponseTests.cs:
```diff
@@ -0,0 +1,178 @@
+    }
+
+	[Fact]
+    public void GetTodoPOSTValidResponseNoBodyTest()
```
Good name (the NoBody made it obvious for me what you were testing).

evan-m-jackson replied:
Thanks!

evan-m-jackson committed:
Add acceptance test commands to Test YAML file

evan-m-jackson committed:
Add acceptance test commands to Test YAML file

paytonrules commented on HTTPServerProject/Program.cs:
```diff
@@ -40,34 +48,53 @@ public void Conversation()
 
             var httpType = header.GetRequestType(initialLine);
             var httpPath = header.GetPath(initialLine);
+            
```
Got some indentation issues here.

evan-m-jackson replied:
Agreed. Will update

paytonrules reviewed (commented):
Not done - but I realized I was focusing too much on micro and want to step back and look at this at a higher level, so let me get these comments in now.

evan-m-jackson committed:
Merge branch 'update-features' of https://github.com/evan-m-jackson/HTTP-Server into update-features

evan-m-jackson committed:
Separate acceptance test step into different steps

evan-m-jackson committed:
Test YAML

evan-m-jackson committed:
Update YAML file

evan-m-jackson committed:
Update Ruby uses step in YAML file

evan-m-jackson committed:
Format code and update Test YAML file

- Add named parameters to functions with more than 2 parameters.
- Add multiple lines for Lists with more than 2 items
- Remove acceptance tests for proxy server in Test YAML file
- Fix indentation

evan-m-jackson committed:
Test YAML file

evan-m-jackson committed:
Update Rakefile to return error code for tests

evan-m-jackson committed:
Add separate rake task for Acceptance Tests in Rakefile

evan-m-jackson committed:
Refactor 'Conversation' method

- Refactor 'Conversation' to have port number as parameter, set default to 5000.
- Refactor 'PathParameters' class to have port number as parameter in order to make '/redirect' path more flexible
= Refactor 'TestPathParameters' class to have default port number of 5000 to make tests pass

evan-m-jackson committed:
Refactor Test YAML file

- Shutdown server in 'Test' step
